src/ddmd/parse.d

```diff
@@ -4766,7 +4766,14 @@ final class Parser(AST) : Lexer
             }
             else if (!f.frequire && !f.fensure) // allow these even with no body
             {
-                error("semicolon expected following function declaration");
+                TOK t = token.value;
+                if (t == TOKconst || t == TOKimmutable || t == TOKinout || t == TOKreturn ||
+                    t == TOKshared || t == TOKnothrow || t == TOKpure)
+                    error("'%s' cannot be placed after a template constraint", token.toChars);
+                else if (t == TOKat)
+                    error("annotations cannot be placed after a template constraint");
```

Comment:
rather _attributes_ no ? _annotation_  isn't an official name.

Reply:
yeah, it does sound better. thanks!